src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -126,6 +126,7 @@ internal partial class Binder
             Debug.Assert(!InExecutableBinder); // Cannot eagerly report diagnostics handled by LazyMissingNonNullTypesContextDiagnosticInfo 
             bool hasTypeLikeConstraint = false;
             bool reportedOverrideWithConstraints = false;
+            MessageID? firstTypeConstraintString = null;
```

Comment:
>MessageID? firstTypeConstraintString = null; [](start = 12, length = 44)

The implementation feels unnecessary complicated. I suggest that we simply report a message that says: "The 'class', 'struct', 'unmanaged' and 'notnull' constraints cannot be repeated or combined with each other and should be specified first." All that complicated tracking system will be unnecessary and we can easily ensure that this error is reported only once.